modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClayManagementToolbar from '@clayui/management-toolbar';
+import ClayModal from '@clayui/modal';
+import ClayTable from '@clayui/table';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const TranslationModal = ({
+	ariaLabels = {
+		default: Liferay.Language.get('default'),
+		manageTranslations: Liferay.Language.get('manage-translations'),
+		translated: Liferay.Language.get('translated'),
+		untranslated: Liferay.Language.get('untranslated'),
+	},
+	defaultLocale,
+	locales,
+	observer,
+	translations,
+}) => {
+	const [active, setActive] = React.useState(false);
+	const [icon, setIcon] = React.useState('search');
+	const inputRef = React.createRef();
+	const [visibleLocales, setVisibleLocales] = React.useState(locales);
+
+	return (
+		<ClayModal observer={observer}>
+			<ClayModal.Header>
+				{Liferay.Language.get('manage-translations')}
+			</ClayModal.Header>
+
+			<ClayModal.Body scrollable>
+				<ClayManagementToolbar aria-label="Management toolbar">
+					<ClayManagementToolbar.Search showMobile={true}>
+						<ClayInput.Group>
+							<ClayInput.GroupItem>
+								<ClayInput
+									aria-label={Liferay.Language.get('search')}
+									className="form-control input-group-inset input-group-inset-after"
+									onInput={() => {
+										const value = inputRef.current?.value;
```

Comment:
How come you went with a ref instead of using the event.target?

```jsx
onInput={(event) => {
    const {value} = event.target;
    //...
}}

Reply:
That's a good question ... I actually used a ref, because I thought that was the React way to do things, but using event.target did come to mind. I'll use that next time, thanks

Reply:
Updated in [`8a9dfad`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/8a9dfad58eeeec19dfd11eed9e325170680b3266) - sorry this might be the "hardest" thing to review, but it does look nicer

Reply:
Typically the "react way"™️  is to use [Controlled Components](https://reactjs.org/docs/forms.html#controlled-components). Although, there are cases where controlled components cause issues, so I have actually seen people use refs to get around those issues.

More thoughts on controlling inputs... Additionally, controlling an input isn't _always_ necessary. You really only need to control it when you want to do something based on that state. For example, if you just have a simple form you can leave all the inputs uncontrolled and just get the data on form submit. However, if you want to render something based on the value of one of those inputs then you need to control it. See below for an example or [codesandbox](https://codesandbox.io/s/trusting-mendel-t4uzh?file=/src/App.js).

```jsx
const UncontrolledComponent = () => {
  function handleSubmit(event) {
    event.preventDefault();

    const data = {
      firstName: event.target.firstName.value,
      lastName: event.target.lastName.value,
      age: event.target.age.value
    };

    // POST your data here
  }

  return (
    <form onSubmit={handleSubmit}>
      <input name="firstName" />
      <input name="lastName" />
      <input name="age" type="number" />
      <button>Submit</button>
    </form>
  );
};

// Note below how we want to update the submit button based on the value of the inputs.

const ControlledComponent = () => {
  const [firstName, setFirstName] = React.useState("");
  const [lastName, setLastName] = React.useState("");
  const [age, setAge] = React.useState("");

  function handleSubmit(event) {
    event.preventDefault();

    const data = {
      firstName,
      lastName,
      age
    };

    // POST your data here
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        name="firstName"
        onChange={(event) => setFirstName(event.target.value)}
        value={firstName}
      />
      <input
        name="lastName"
        onChange={(event) => setLastName(event.target.value)}
        value={lastName}
      />
      <input
        name="age"
        onChange={(event) => setAge(event.target.value)}
        type="number"
        value={age}
      />
      <button disabled={!firstName.length || !lastName.length || !age.length}>
        Submit
      </button>
    </form>
  );
};
```

So for your specific case, it is helpful to control the input because you are filtering based on the input.

Reply:
Thanks, once again that's a very helpful explanation and code example. Makes much more sense now.